src/react/index.js

```diff
@@ -37,25 +49,62 @@ export default function (tag, cls, vars = [], content) {
     }
 
     render () {
-      const { props, state } = this
+      const { props, state, context } = this
       const mergedProps = {
         ...props,
         theme: state.theme
       }
 
-      const getValue = v =>
-        v && typeof v === 'function' ? v.cls || v(mergedProps) : v
+      const getValue = v => {
+        if (v && typeof v === 'function') {
+          if (v.__emotion_spec) {
+            return css(
+              map(v.__emotion_spec.objs, getValue),
+              map(v.__emotion_spec.vars, getValue),
+              v.__emotion_spec.content
+            )
```

Comment:
I don't understand this, wouldn't this break with components that have function interpolations? e.g. in this example the class generated to target the child selector would be based on the `fontSize` prop of `AnotherComponent`
```jsx
const SomeComponent = styled.div`
  font-size: ${props => props.fontSize || 20};
  display: flex;
`

const AnotherComponent = styled.div`
  ${SomeComponent} {
    border: 0;
  }
`
```